src/components/Calculator.js

```diff
@@ -1,28 +1,80 @@
-export const Calculator = () => (
-  <div className="bg-white w-2/6 h-2/4 rounded oveflow-hidden">
-    <div className="bg-gray-500 h-1/6 flex items-center justify-end p-5">
-      <p className="text-2xl text-white">0</p>
+import React, { useState, useEffect } from 'react';
+import calculate from '../logic/calculate';
+
+export const Calculator = () => {
+  const [firstValue, setFirstValue] = useState('');
+  const [secondValue, setSecondValue] = useState('');
+  const [operator, setOperator] = useState('');
+  const [dataObject, setDataObject] = useState();
+  const [result, setResult] = useState('');
+
+  const operators = '+-x÷%';
+  const numbers = '0123456789';
+
+  const handleValues = (event) => {
+    if (numbers.includes(event.target.name)) {
+      if (firstValue === '') {
+        setFirstValue(event.target.name);
+        setResult(`${result}${event.target.name}`);
+      } else {
+        setSecondValue(event.target.name);
+        setResult(`${result}${event.target.name}`);
+      }
+    }
+
+    if (operators.includes(event.target.name)) {
+      setOperator(event.target.name);
+      setResult(`${result}${event.target.name}`);
+    }
+
+    const response = calculate(dataObject, event.target.name);
+    if (response) {
+      setFirstValue(response.total);
+      setSecondValue(response.next);
+      setOperator(response.operation);
+
+      if (response.next === null && response.operation === null) {
+        setResult(`${response.total || ''}`);
+      }
+    }
+  };
+
+  useEffect(() => {
+    const newDataObject = {
+      total: firstValue,
+      next: secondValue,
+      operation: operator,
+    };
+   
+    setDataObject(newDataObject);
+  }, [firstValue, secondValue, operator]);
+
+  return (
+    <div className="bg-white w-2/6 h-2/4 rounded oveflow-hidden">
+      <div className="bg-gray-500 h-1/6 flex items-center justify-end p-5">
+        <p className="text-2xl text-white">{ result }</p>
+      </div>
+      <div className="bg-gray-400 h-5/6 grid grid-cols-4 gap-1">
+        <button name="AC" onClick={handleValues} className="bg-white w-22">AC</button>
+        <button name="+/-" onClick={handleValues} className="bg-white w-22">+/-</button>
```

Comment:
- [ ] Can you please add visuals on the calculator for the `+/-` and ` . ` buttons? For example, if a user types `-5 + 2`, he rather sees `5 + 2` but it gives the correct answer `-3`